### PR TITLE
PB-7691: Add support for Azure immutable container creation

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:
- [x] Add func `GetAzureImmutabilityCredsFromEnv()` to fetch Immutability specific ENVs. 
- [x] Add new ENVs to `deployments/deploy-pxb-cloud.sh` and `deployments/deploy-ssh.sh`
- [x] Update  func`CreateAzureBucket()` to incorporate new parameters. 
- [x] Update func `CreateBucket()` to use new default parameters. 

**Why are we not using Go client and going for this command based approach?** 
I had a look at the Azure go client where I saw it was a multi staged client where we don't have support to create an container with version enabled. And the APIs to set to policy is is also in a different client based on which the cli command code it much cleaner approach for now once Azure go client supports all the calls we can update the code to use it.  

**Successful run:**
Have updated the test `BackupClusterVerification` with the function calls will remove before merging and given it a run: 
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/azure-immutablity-byoc-test/12/console
~https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/azure-immutablity-byoc-test/10/console~

**Which issue(s) this PR fixes** (optional)
Closes #PB-7691

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/common.go [hunk 1]
```diff
@@ -7019,29 +7019,120 @@ func RemoveS3BucketPolicy(bucketName string) error {
...
+	expect(containerLevelSAKey).NotTo(equal(""),
+		"AZURE_STORAGE_ACCOUNT_KEY_CONTAINER_LEVEL Environment variable should not be empty")
+
+	storageAccountLevelSA = os.Getenv("AZURE_STORAGE_ACCOUNT_NAME_SA_LEVEL")
+	expect(storageAccountLevelSA).NotTo(equal(""),
+		"AZURE_STORAGE_ACCOUNT_NAME_SA_LEVEL Environment variable should not be empty")
+
+	storageAccountLevelSAKey = os.Getenv("AZURE_STORAGE_ACCOUNT_KEY_SA_LEVEL")
+	expect(storageAccountLevelSAKey).NotTo(equal(""),
+		"AZURE_STORAGE_ACCOUNT_KEY_CONTAINER_LEVEL Environment variable should not be empty")
```
Comment (kphalgun-px) [inline tests/common.go:7042]: AZURE_STORAGE_ACCOUNT_KEY_SA_LEVEL Environment variable should not be empty?

Comment (kshithijiyer-px) [reply tests/common.go:7042]: Yes we need to make sure it's not empty.

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/common.go [hunk 2]
```diff
@@ -7019,29 +7019,120 @@ func RemoveS3BucketPolicy(bucketName string) error {
...
+				cmd = fmt.Sprintf("az cloud set --name AzureChinaCloud")
+				_, stdErr, err = osutils.ExecShell(cmd)
+				expect(err).NotTo(haveOccurred(),
+					fmt.Sprintf("Failed to set cloud account to AzureChina. Error: [%v] [%v]", err, stdErr))
+			}
+			cmd = fmt.Sprintf("az storage container-rm create --name %s  --storage-account %s --resource-group %s "+
+				"--enable-vlw", bucketName, containerLevelSA, resourceGroup)
+			_, stdErr, err = osutils.ExecShell(cmd)
+			expect(err).NotTo(haveOccurred(),
+				fmt.Sprintf("Failed to create bucket with Verioning Enabled. Error: [%v] [%v]", err, stdErr))
```
Comment (ak-px) [inline tests/common.go:7080]: nit: versioning

File: tests/common.go [hunk 3]
```diff
@@ -7019,29 +7019,120 @@ func RemoveS3BucketPolicy(bucketName string) error {
...
+
+			cmd = fmt.Sprintf("az storage container immutability-policy create "+
+				"--resource-group %s --account-name %s --container-name %s --period %d",
+				resourceGroup, containerLevelSA, bucketName, retentionDays)
+			_, stdErr, err = osutils.ExecShell(cmd)
+			expect(err).NotTo(haveOccurred(),
+				fmt.Sprintf("Failed to enaled immutability-policy on the bucket. Error: [%v] [%v]", err, stdErr))
```
Comment (ak-px) [inline tests/common.go:7087]: nit enable

**Review by ak-px — APPROVED** [review]
LGTM, minor comments.

File: tests/common.go (hunk 2 above)
Comment (kshithijiyer-px) [reply tests/common.go]: Done

File: tests/common.go (hunk 3 above)
Comment (kshithijiyer-px) [reply tests/common.go]: Done

File: tests/common.go [hunk 4]
```diff
@@ -7019,29 +7019,120 @@ func RemoveS3BucketPolicy(bucketName string) error {
...
+	safeAccountLevelSAKey = os.Getenv("AZURE_STORAGE_ACCOUNT_KEY_SA_LEVEL_SAFE")
+	expect(safeAccountLevelSAKey).NotTo(equal(""),
+		"AZURE_STORAGE_ACCOUNT_KEY_SA_LEVEL_SAFE Environment variable should not be empty")
+
+	return resourceGroup, containerLevelSA, containerLevelSAKey, storageAccountLevelSA, storageAccountLevelSAKey, safeAccountLevelSA, safeAccountLevelSAKey
+}
+
 // CreateAzureBucket creates bucket in Azure
-func CreateAzureBucket(bucketName string) {
+func CreateAzureBucket(bucketName string, objectlock bool, mode string, retentionDays int, safeMode bool) {
```
Comment (mkoppal-px) [inline tests/common.go:7056]: Make the `mode` as a type instead of string

**Review by mkoppal-px — APPROVED** [review]
LGTM except for a minor comment

File: tests/common.go (hunk 4 above)
Comment (kshithijiyer-px) [reply tests/common.go]: Done

**Pull request merged**